app/controllers/application_controller.rb

```diff
@@ -1,2 +1,18 @@
 class ApplicationController < ActionController::API
+  # https://github.com/gonzalo-bulnes/simple_token_authentication#make-models-token-authenticatable
+  acts_as_token_authentication_handler_for User, except: [:index, :show, :linkedin], fallback: :exception
+
+  private
+
```

Comment:
So my concern with this approach to checking privileges, is that a fair amount of our endpoints will be used by all three types of user roles. For example, to create a new medium, the request may be for a user's profile, an organization's profile, or a program's profile. This means that if we want to restrict access using the before_action callback, then whatever method we have it call will have to make use of the super_admin_only and admin_only helper methods internally based on further logic instead of just calling these helpers directly from the before_action. Now, this may be perfectly fine, but I wanted to bring it up anyway and discuss it with you guys. The alternative is making checks against user role in the actual action method itself depending on what type of parameters are given. Does that all make sense?

Reply:
Let's talk about this in a meeting later this week